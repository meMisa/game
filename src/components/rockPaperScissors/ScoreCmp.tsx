// node modules
import * as React from "react";
// assets :
import Logo_Bonus from "assets/images/logo-bonus.svg";
// constants
import { LOGO_BONUS_ALT, SCORE } from "constants/texts";
// styles
import "styles/score.css";

const ScoreCmp = (props: { score: number }) => (
  <div
    className="justify-content-space-between score-wrapper"
    data-testid="score"
  >
    <div className="score-logo-bonus">
      <img src={Logo_Bonus} alt={LOGO_BONUS_ALT} />
    </div>
    <div className="score-text score-bonus-text-container">
      <p>{SCORE}</p>
      <div className="dark-text count">
        <span>{props.score}</span>
      </div>
    </div>
  </div>
);
export default ScoreCmp;
