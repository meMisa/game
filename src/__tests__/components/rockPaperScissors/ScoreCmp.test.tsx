import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

import ScoreCmp from "components/rockPaperScissors/ScoreCmp";

const score = 0;

afterEach(() => {
  cleanup();
});

describe("Test", () => {
  test("should render ScoreCmp component correctly", () => {
    render(<ScoreCmp score={score} />);
    const scoreComponent = screen.getByTestId("score");
    expect(scoreComponent).toBeInTheDocument();
    expect(scoreComponent).toHaveTextContent("SCORE");
  });

  test("match snapshot", () => {
    const tree = renderer.create(<ScoreCmp score={3} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
