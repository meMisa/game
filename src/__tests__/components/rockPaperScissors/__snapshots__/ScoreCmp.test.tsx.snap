// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test match snapshot 1`] = `
<div
  className="justify-content-space-between score-wrapper"
  data-testid="score"
>
  <div
    className="score-logo-bonus"
  >
    <img
      alt="Rock, Paper, Scissors, Lizard, Spock"
      src="logo-bonus.svg"
    />
  </div>
  <div
    className="score-text score-bonus-text-container"
  >
    <p>
      SCORE
    </p>
    <div
      className="dark-text count"
    >
      <span>
        3
      </span>
    </div>
  </div>
</div>
`;
